src/components/Avatar/Addon.js

```diff
@@ -0,0 +1,35 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import style from './style.css';
+
+const Addon = ({ position, offset, ...rest }) => {
+  const positions = useMemo(() => {
+    const p = position.split('-');
+    return {
+      [p[0]]: `${offset}px`,
+      [p[1]]: `${offset}px`,
+    };
+  }, [position, offset]);
+
+  const badgeClassNames = classNames(style.AvatarBadge, rest.className);
+  return <div {...rest} className={badgeClassNames} style={positions} />;
```

Comment:
If you keep the `positions` props, would you change this to ...
```
{{
  ...positions,
  rest.style,
}}
```
... so the consumer doesn't lose the ability to customize or override styles.